src/status_im2/contexts/wallet/receive/view.cljs

```diff
@@ -0,0 +1,102 @@
+(ns status-im2.contexts.wallet.receive.view
+  (:require
+    [quo.core :as quo]
+    [react-native.blur :as blur]
+    [react-native.core :as rn]
+    [react-native.platform :as platform]
+    [react-native.safe-area :as safe-area]
+    [react-native.share :as share]
+    [reagent.core :as reagent]
+    [status-im2.common.qr-codes.view :as qr-codes]
+    [status-im2.constants :as constants]
+    [status-im2.contexts.wallet.common.sheets.network-preferences.view :as network-preferences]
+    [status-im2.contexts.wallet.receive.style :as style]
+    [utils.i18n :as i18n]
+    [utils.image-server :as image-server]
+    [utils.re-frame :as rf]))
+
+(def id-to-network
+  {constants/mainnet-chain-id  :ethereum
+   constants/optimism-chain-id :optimism
+   constants/arbitrum-chain-id :arbitrum})
+
+(defn- share-action
+  [address share-title]
+  (share/open
+   (if platform/ios?
+     {:activityItemSources [{:placeholderItem {:type    "text"
+                                               :content address}
+                             :item            {:default {:type "text"
+                                                         :content
+                                                         address}}
+                             :linkMetadata    {:title share-title}}]}
+     {:title   share-title
+      :subject share-title
+      :message address})))
+
+(defn- open-preferences
+  [selected-networks]
+  (rf/dispatch [:show-bottom-sheet
+                {:content
+                 (fn []
+                   [network-preferences/view
+                    {:blur?             true
+                     :selected-networks (set @selected-networks)
+                     :on-save           (fn [chain-ids]
+                                          (rf/dispatch [:hide-bottom-sheet])
+                                          (reset! selected-networks (map #(get id-to-network %)
+                                                                         chain-ids))
+                                        )}])}]))
+
+
+(defn view
+  []
+  (let [padding-top       (:top (safe-area/get-insets))
+        wallet-type       (reagent/atom :wallet-legacy)
+        selected-networks (reagent/atom [:ethereum :optimism :arbitrum])]
+    (fn []
+      (let [{:keys [address color emoji] :as account} (rf/sub [:wallet/current-viewing-account])
+            share-title                               (str (:name account) " " (i18n/label :t/address))
```

Comment:
this i18n label should take this as a prop. imo we want to assume as little as possible about the translated language and give the translator control of this.
i.e `(i18n/label :t/address {:name  (:name account)})`

Reply:
@J-Son89 I think this would try to translate the account name? We probably don't want to do that, we want to display the account name as it is

Reply:
Indoor believe so, it would just give a translator the ability of where to put the account name in a sentence.

E,g
Jamie Account or Account Jamie

Reply:
@J-Son89 I am saying that `(i18n/label :t/address {:name  (:name account)})` would also translate the account name, right? For example if the account name is something like "black horse", then it would translate, but we probably don't want to do that, right?

Reply:
tbh I couldn't see in the UI how this looks, but my thought is this.
For the  title `(str (:name account) " " (i18n/label :t/address)` if the output is (for eg) "Omar Address" then we should avoid assuming how it will look in other languages and leave the option available to change it.
in this example so it could be "Address Omar" or any other order. Also the name Omar would not be translatable but just accessible to use in the translation string.
It's only a nit so no need to adjust if you don't think it makes sense here, but just in general we should leave as much control to the translation as possible imo.